client/blocks/timezone/index.jsx

```diff
@@ -0,0 +1,95 @@
+/**
+ * External Dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+
+import { map, toPairs } from 'lodash';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import QueryTimezones from 'components/data/query-timezones';
+import { timezonesRequestAction as requestTimezones } from 'state/timezones/actions';
+import {
+	getRawOffsets,
+	getTimezones,
+} from 'state/selectors';
+
+/**
+ * Module variables
+ */
+const noop = () => {};
+
+class Timezone extends Component {
+	static propTypes = {
+		selectedZone: PropTypes.string,
+		onSelect: PropTypes.func
+	};
+
+	static defaultProps = {
+		selectedZone: '',
+		onSelect: noop
+	};
+
+	selectTimezone = ( { target } ) => {
+		this.props.onSelect( target.value );
```

Comment:
To be safe, let's use lodash `get( target, 'value' )`, or `target && target.value`